python/cuspatial/cuspatial/core/binpreds/feature_contains.py

```diff
@@ -0,0 +1,369 @@
+        )
+        # Because the quadtree contains_properly call returns a list of
+        # points that are contained in each part, parts can be duplicated
+        # once their index is converted to a polygon index.
+        allpairs_result = polygon_indices.drop_duplicates()
+
+        # Replace the polygon index with the original index
+        allpairs_result["polygon_index"] = allpairs_result[
+            "polygon_index"
+        ].replace(Series(lhs.index, index=cp.arange(len(lhs.index))))
+
+        return allpairs_result
+
+    def _postprocess(self, lhs, rhs, op_result):
+        """Postprocess the output GeoSeries to ensure that they are of the
+        correct type for the predicate.
+
+        Postprocess for contains_properly has to handle multiple input and
+        output configurations.
+
+        The input can be a single polygon, a single multipolygon, or a
+        GeoSeries containing a mix of polygons and multipolygons.
+
+        The input to postprocess is `point_indices`, which can be either a
+        cudf.DataFrame with one row per point and one column per polygon or
+        a cudf.DataFrame containing the point index and the part index for
+        each point in the polygon.
+
+        Parameters
+        ----------
+        lhs : GeoSeries
+            The left-hand side of the binary predicate.
+        rhs : GeoSeries
+            The right-hand side of the binary predicate.
+        preprocessor_output : ContainsOpResult
+            The result of the contains_properly call.
+
+        Returns
+        -------
+        cudf.Series or cudf.DataFrame
+            A Series of boolean values indicating whether each feature in
+            the rhs GeoSeries is contained in the lhs GeoSeries in the
+            case of allpairs=False. Otherwise, a DataFrame containing the
+            point index and the polygon index for each point in the
+            polygon.
+        """
+        if len(op_result.result) == 0:
+            return cudf.Series([False] * len(lhs))
+
+        # Convert the quadtree part indices df into a polygon indices df.
+        # Helps with handling multipolygons.
+        allpairs_result = self._reindex_allpairs(lhs, op_result)
+
+        # If the user wants all pairs, return the result. Otherwise,
+        # return a boolean series indicating whether each point is
+        # contained in the corresponding polygon.
+        if self.config.allpairs:
+            return allpairs_result
+        else:
+            # for each input pair i: result[i] =  true iff point[i] is
+            # contained in at least one polygon of multipolygon[i].
+            # pairwise
+            if len(lhs) == len(rhs):
+                matches = (
+                    allpairs_result["polygon_index"]
+                    == allpairs_result["point_index"]
+                )
+                polygon_indexes = allpairs_result["polygon_index"][matches]
+                final_result = Series([False] * len(rhs))
```

Comment:
Probably better to create this on device:
```suggestion
                final_result = Series(cupy.zeros(len(rhs), dtype=bool))
```

Reply:
Indeed. Because this semantic is used so frequently I've moved them all to a `binpred_utils.py::_false_series(len)` function.